src/app/InteractionModelHelper.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <system/SystemPacketBuffer.h>
+namespace chip {
+namespace app {
+
+static CHIP_ERROR ReserveBuffer(System::PacketBufferTLVWriter & aWriter, uint32_t aReserveSpace)
```

Comment:
Why a separate header for this?

But in any case, this should be called something like `InitWriterWithSpaceReserved` or something.

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/src/app/WriteClient.cpp#L227 has used similar code, I possibly would update this code to use the same function in follow-up.

update the naming, thanks